src/write/unit.rs

```diff
@@ -248,6 +248,36 @@ impl Unit {
         self.root
     }
 
+    /// Reserve a `DebuggingInformationEntry` in this unit and return its id.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `parent` is invalid.
+    pub fn reserve(&mut self) -> UnitEntryId {
+        DebuggingInformationEntry::new(
+            self.base_id,
+            &mut self.entries,
+            None,
+            constants::DW_TAG_null,
+        )
+    }
+
+    /// Set the parent and tag of a previously reserved `DebuggingInformationEntry`.
+    ///
+    /// The `parent` must be within the same unit.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `child` or `parent` is invalid, or if `child` is not a reserved entry.
+    pub fn add_reserved(&mut self, child: UnitEntryId, parent: UnitEntryId, tag: constants::DwTag) {
+        let entry = self.get_mut(child);
+        debug_assert_eq!(entry.parent, None);
+        debug_assert_eq!(entry.tag, constants::DW_TAG_null);
+        entry.parent = Some(parent);
+        entry.tag = tag;
+        self.get_mut(parent).children.push(child);
+    }
```

Comment:
I think `parent` is just here. Was the previous `parent` reference just out-of-sync with how you ended up structuring the APIs?